social_core/backends/azuread.py

```diff
@@ -83,32 +80,34 @@ def get_user_id(self, details, response):
         return response.get("upn")
 
     def get_user_details(self, response):
-        """Return user details from Azure AD account"""
-        fullname, first_name, last_name = (
-            response.get("name", ""),
-            response.get("given_name", ""),
-            response.get("family_name", ""),
-        )
+        """Return user details from Microsoft online account"""
+        email = response.get("mail")
+        username = response.get("userPrincipalName")
+
+        if "@" in username:
+            if not email:
+                email = username
+            username = username.split("@", 1)[0]
+
         return {
-            "username": fullname,
-            "email": response.get("email", response.get("upn")),
-            "fullname": fullname,
-            "first_name": first_name,
-            "last_name": last_name,
+            "username": username,
+            "email": email,
```

Comment:
`email` used to fall back to `userPrincipalName` if not present, this is now gone.